src/Core/Grid/Query/CustomerQueryBuilder.php

```diff
@@ -222,7 +222,9 @@ private function applyFilters(array $filters, QueryBuilder $qb)
                 continue;
             }
 
-            $qb->andWhere('c.`' . $filterName . '` LIKE :' . $filterName);
+            $customerTableAlias = in_array($filterName, ['firstname', 'lastname', 'company', 'email']);
+
+            $qb->andWhere(($customerTableAlias ? 'c.' : '') . '`' . $filterName . '` LIKE :' . $filterName);
```

Comment:
Mmmm, I know I arrive probably late in this conversation But I don't understand why we remove the alias from the core query so that it works correctly with the module override?
SHouldn't it be the module which needs to adapt? Besides removing an alias can only lead to even more ambiguity, how can it lessen it?